File: app/src/main/java/com/mnj/java_solutions/NumberOfPairsWithAbsoluteDifferenceKLC2006.java
package com.mnj.java_solutions;

public class NumberOfPairsWithAbsoluteDifferenceKLC2006 {

    public void findPairs() {

        int[] a = {1, 2, 2, 1};
        int k = 1;

        int count = 0;
        int[] arr = new int[201];

        for (int j : a) {
            arr[j]++;
        }

        for (int i = k + 1; i < arr.length; i++) {
            count = count + arr[i] * arr[i - k];
        }
        System.out.println("==>> Number of pairs :" + count);
    }

}

File: app/src/main/java/com/mnj/java_solutions/NumberOfPairsWithAbsoluteDifferenceKLC2006.kt
package com.mnj.java_solutions

class NumberOfPairsWithAbsoluteDifferenceKLC2006 {
    fun findPairs() {
        val a = intArrayOf(1, 2, 2, 1)
        val k = 1
        var count = 0
        val arr = IntArray(201)
        for (j in a) {
            arr[j]++
        }
        for (i in k + 1 until arr.size) {
            count = count + arr[i] * arr[i - k]
        }
        println("==>> Number of pairs :$count")
    }
}